From 25d386fc43e560ae2438d49580498b50efd9b534 Mon Sep 17 00:00:00 2001
From: Chris Burroughs <chris.burroughs@joyent.com>
Date: Thu, 31 Jan 2019 16:10:58 -0500
Subject: [PATCH] TRITON-1192 setup-grafana.sh should be less lame; retry to
 see if grafana is up Reviewed by: Isaac Davis <isaac.davis@joyent.com>
 Approved by: Isaac Davis <isaac.davis@joyent.com>

---
 setup-grafana.sh | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/setup-grafana.sh b/setup-grafana.sh
index 3fc09df..f09ede0 100755
--- a/setup-grafana.sh
+++ b/setup-grafana.sh
@@ -181,9 +181,15 @@ SYSTEMD
 
 zlogin \${vm_uuid} "systemctl daemon-reload && systemctl enable grafana && systemctl start grafana && systemctl status grafana" </dev/null
 
-# Lame method to allow grafana to start and provision the dashboards.
-# Would be better to poll the curl attempts below.
 sleep 5
+retries=10
+while [[ \${retries} -gt 0 ]]; do
+    if curl -sSf -u admin:admin "\${grafana_ip}:3000/api/search?type=dash-db&query=cnapi"; then
+        break;
+    fi
+    let "retries=retries-1"
+    sleep 5
+done
 
 # Set the CNAPI dashboard (for now) as the default org dashboard.
 alias json="/native/usr/node/bin/node /native/usr/bin/json"
-- 
2.21.0

